diff --git a/test/prefetch/Makefile b/test/prefetch/Makefile
index b271f194..322f1331 100644
--- a/test/prefetch/Makefile
+++ b/test/prefetch/Makefile
@@ -60,56 +60,9 @@ urls_and_accs:
         then echo "prefetch URL-2 when there is no kfg should fail"; exit 1; \
 	    fi
 
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; \
-	    if ping -c1 intranet > /dev/null ; then \
-	       if $(BINDIR)/prefetch http://intranet/ 2>/dev/null ; \
-	       then echo "prefetch URL-2 when there is no kfg should fail"; exit 1; \
-	       fi ; \
-	    fi
-
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; \
-	    if ping -c1 intranet > /dev/null ; then \
-	     if $(BINDIR)/prefetch http://intranet/irb/support.shtml  2>/dev/null ; \
-	     then echo "prefetch URL-2 when there is no kfg should fail"; exit 1; \
-	     fi ; \
-	    fi
-
 	@echo '$(PUBLIC)/apps/file/volumes/flat = "files"' > tmp/t.kfg
 	@echo '$(PUBLIC)/root = "$(CWD)/tmp"' >> tmp/t.kfg
 
-	@if ls `pwd`/tmp2/support.shtml 2> /dev/null ; \
-     then echo support.shtml found ; exit 1 ; fi
-	@echo HTTP download when user repository is configured
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; cd tmp2 ; \
-	    if ping -c1 intranet > /dev/null ; then \
-	    $(BINDIR)/prefetch http://intranet/irb/support.shtml > /dev/null ; fi
-	@if ping -c1 intranet > /dev/null ; \
-	 then ls `pwd`/tmp2/support.shtml > /dev/null ; fi
-#	@echo
-
-	@echo Running prefetch second time finds previous download
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; cd tmp2 ; \
-	 if ping -c1 intranet > /dev/null ; then \
-	     $(BINDIR)/prefetch http://intranet/irb/support.shtml \
-	        | grep "found local" > /dev/null ; fi
-#	@echo
-
-	@if ls `pwd`/tmp2/index.html 2> /dev/null ; \
-     then echo index.shtml found ; exit 1; fi
-	@echo HTTP download when user repository is configured
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; cd tmp2 ; \
-	 if ping -c1 intranet > /dev/null ; then \
-	    $(BINDIR)/prefetch http://intranet/ > /dev/null ; fi
-	@if ping -c1 intranet>/dev/null; then ls `pwd`/tmp2/index.html > /dev/null;fi
-#	@echo
-
-	@echo Running prefetch second time finds previous download
-	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; cd tmp2 ; \
-	 if ping -c1 intranet > /dev/null ; then \
-	    $(BINDIR)/prefetch http://intranet/ | grep "found local" > /dev/null ; fi
-	@rm -f tmp2/index.html
-#	@echo
-
 	@echo URL download when user repository is configured
 	@if ls `pwd`/tmp2/wiki 2> /dev/null ; then echo wiki found ; exit 1; fi
 	@export VDB_CONFIG=`pwd`/tmp; export NCBI_SETTINGS=/ ; cd tmp2 ; \
