diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6b7ef6..1beba22 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -490,6 +490,8 @@ ExternalProject_Add(libdivsufsort
     #UPDATE_COMMAND sh -c "mkdir -p <SOURCE_DIR>/build"
     BINARY_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/libdivsufsort-master/build
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR> -DBUILD_DIVSUFSORT64=TRUE -DUSE_OPENMP=TRUE -DBUILD_SHARED_LIBS=FALSE
+
+    INSTALL_COMMAND make install
 )
 ExternalProject_Add_Step(libdivsufsort makedir
   COMMAND mkdir -p <SOURCE_DIR>/build 
